Either/Either.swift

```diff
@@ -96,6 +96,38 @@ public func >>- <T, U, V> (either: Either<T, U>, @noescape transform: U -> Eithe
 }
 
 
+///  Extracts from a list of `Either` all the `Left` elements. All the `Left` elements are extracted in order.
+public func lefts <T, U>(list: [Either<T, U>]) -> [T] {
+	return list.reduce([T]()) { ( var memo, e) in
+		if let l = e.left {
+			memo.append(l)
+		}
+		
+		return memo
+	}
+}
+
+
+///  Extracts from a list of `Either` all the `Right` elements. All the `Right` elements are extracted in order.
+public func rights <T, U>(list: [Either<T, U>]) -> [U] {
+	return list.reduce([U]()) { ( var memo, e) in
+		if let l = e.right {
+			memo.append(l)
+		}
+		
+		return memo
+	}
+}
+
+
+///  Partitions a list of `Either` into two lists.
+///
+///  All the `Left` elements are extracted in order to the first component of the output. The `Right` elemests are extracted in order to the second component of the output.
+public func partitionEithers <T, U>(list: [Either<T, U>]) -> ([T], [U]) {
+	return (lefts(list), rights(list))
+}
```

Comment:
Some questions:

1. What do you think about making these `static` functions on `Either` instead of free functions?

2. Alternatively, what do you think about implementing them as free functions over `EitherType` instead of over `Either`?

3. Would it make sense to implement `lefts`/`rights` over arbitrary `SequenceType` instead of requiring `Array`? If we do decide thus, could you please adjust the docs & parameters to `sequence` instead of `list`?

4. Would it make sense to use `SequenceOf`, `FilterSequenceView`, or `MapSequenceView` to make the result lazy?